share/spice/caniuse_lists/caniuse_lists.js

```diff
@@ -0,0 +1,164 @@
+
+            removeHyphen(versions);
+
+            versions.sort(function(a,b) {   //sort descending
+                return b - a;
+            });
+
+            for(var type in support_type) {
+                return_value = findSupportInfo(compatibility, versions, index, type, support_type[type]);
+                if(return_value && return_value['found']) {
+                    minVersion = return_value['minVersion'];
+                    break;
+                } else if(return_value && !return_value['found']) {
+                    index = return_value['index'];
+                }
+            }
+            if(minVersion == '--') {
+                supported[browser_name] = false;
+            }
+            return minVersion;
+        }
+              
+        function findSupportInfo(compatibility, versions, index, type, support_type) {
+            var minVersion,
+                found = false,
+                yp_found = false;
+            //for support values of yes, partial
+            while(support_type.indexOf(compatibility[versions[index]]) != -1) {
+                minVersion = versions[index];
+                minVersion += '+';
+                found = true;
+                index++;
+                yp_found = true;
+            }
+            //for support values of the form 'a #n', 'y #n', etc
+            while(!yp_found && versions[index] && compatibility[versions[index]].match(/([y|a|a x]\s#)+/g)) {
+                minVersion = versions[index];
+                minVersion += '+*';
+                found = true;
+                index++;
+            }
+            if(!found)
+                return;
+            if(type == 'partial') {
+                minVersion += '*';
+            }
+            return {'index': index, 'minVersion': minVersion, 'found': found};   //return how much we have advanced in the array
+        }
+        
+        function removeHyphen(versions) {  //separate versions having a range, eg. 4.4.3-4.4.4
+            var index = 0,
+                split_v;
+            
+            for(index = 0; index < versions.length; index++) {
+                split_v = versions[index].split('-'),
+                split_v = removePeriods(split_v);
+                if(split_v.length > 1)
+                    versions.splice(index, 1, split_v[0], split_v[1]);
+                else
+                    versions.splice(index, 1, split_v[0]);
+            }
+        }
+
+        function removePeriods(versions) {  //change version of form x.y.z -> x.y, because it has to be sorted
+            var count;
+            
+            for(count = 0; count < versions.length; count++) {
+                var matched = versions[count].match(/\./g);
+                if(matched && matched.length >= 2) {
+                    versions[count] = versions[count].substr(0, versions[count].indexOf('.',versions[count].indexOf('.')));
```

Comment:
this is confusing. What about using a regex to grab the first two version numbers?

Something like `m/^(\d+\.\d+).+$/`?